### ISSUE TITLE
Macros in Batch/External Tracking UI

### ISSUE THREAD

**Issue by allenleetc** [kristinbranson/APT#346]
Enhancements for macro functionality used to specify trkfile outputs in Track>Track Multiple Videos UI:

* Predefined macros should be consistent with those used elsewhere in APT (eg under File>[Export trkfiles]): $movfile, $movdir, $projfile, $projname, $trackertype rather than $name, $dir, $path
* Editbox for "Define trkfile using macros" can be pre-populated with default export trkfile which is currently $movdir/$movfile_$projfile_$trackertype
* User-defined macros could be supported although this is less crucial as these are less commonly used.

**Comment by arobie** [comment]
Editbox for "Define trkfile using macros" can be pre-populated with default export trkfile which is currently $movdir/$movfile_$projfile_$trackertype

This doesn't happen for me on develop? Am I missing something? I still find this very hard to use without default trk naming as a starting point.

**Comment by allenleetc** [comment]
Yes you are right, hand-entering macros into the "Define trk using macros" box doesn't work for me either. The issue description implies that it would/might. 

I don't remember exactly what I tested in terms of macro-functionality on the first pass, but it wasn't much -- basically this issue is a placeholder for all macro-related functionality in that UI. I'll make another pass on this. 

What's the most important core issue here? I am guessing the main pain is clicking or typing a full path to the output trkfile, which for your workflow is a deeply nested file and eg adjacent to the movie-to-be-tracked. If so, I'm not  sure we need the "Define trk using macros" field in the "outer" UI? We could just have the "inner"/detail UI be prepopulated with the default macro (and have that macro work) and we would be good?

**Comment by arobie** [comment]
Yes, if I'm reading correctly, prepopulating with the default trk in the inner GUI would be perfect for me! 

If you want to be more general: 
Maybe instead of the macros in the outer field you could just show the default trk file name and people could edit that if they wanted, and whatever was in that field would auto populate inner trk full path with movie path? 

Have you seen users organizing their data in the 'other' way? With unique movie names? How does auto trk work for that work flow?

**Comment by roian** [comment]
ran into this myself.  i would prefer a pre-populated trk file named based on the movie file, or some way to know a) what macros are available and b) what their current values are.  

also, small GUI issue:  the "add movie" button shows below the interactive gui for adding a specific movie, and i keep clicking it instead of the "done" button.  either grey out while adding a movie, or have the sub-window cover it?

**Comment by allenleetc** [comment]
-  Default trkfile now pre-populated with APT default. This remains in the "outer" UI as it applies to all movies -- basically the @arobie Feb9 solution I believe. 
- Available macros and their values are shown when hovering over the default trkfile label or editbox. This could be made even more explicit, open to suggestions. 
- For projs-with-trx, default trxfile editbox now available; this behaves like the default trkfile.
- Macro handling should be friendlier in general. Basically, set the default trkfile (and trxfile if appropriate) and then just find/type your movies-to-track. The trk/trxfiles should be autofilled.
- Misc minor enhancements.
